actionpack/test/controller/params_wrapper_test.rb

```diff
@@ -79,6 +79,14 @@ def test_store_accessors_wrapped
     end
   end
 
+  def test_internal_parameters_are_filtered_out
+    with_default_wrapper_options do
+      @request.env["CONTENT_TYPE"] = "application/json"
+      post :parse, params: { "username" => "sikachu", "action" => "something" }
+      assert_equal({ "controller" => "params_wrapper_test/users", "action" => "parse", "username" => "sikachu", "user" => { "username" => "sikachu" } }, @request.filtered_parameters)
```

Comment:
`"action" => "something"` should be inside `params["user"]` in the controller and also in the logs. Is it there?